backfill/backfill_manager.py

```
@@ -0,0 +1,474 @@
+    # Run backfill
+    stats = await manager.backfill_from_directory(
+        petition_dir=args.petition_dir,
+        start_year=args.start_year,
+        end_year=args.end_year,
+        chapter_filter=args.chapter,
+        batch_size=args.batch_size
+    )
+    
+    # Save statistics
+    stats_file = f"backfill_stats_{datetime.now().strftime('%Y%m%d_%H%M%S')}.json"
+    with open(stats_file, 'w') as f:
```

Ruff (0.11.9)

Async functions should not open files with blocking methods like `open`

(ASYNC230)